pandas/tests/test_groupby.py

```diff
@@ -5141,6 +5141,118 @@ def test_cumcount_groupby_not_col(self):
         assert_series_equal(expected, g.cumcount())
         assert_series_equal(expected, sg.cumcount())
 
+    def test_enumerate(self):
```

Comment:
can u out these in a separate class TestCounting or something (and maybe move cumcount there as well)